service_user.go

```diff
@@ -78,6 +85,24 @@ func (h *ServiceUsersHandler) Get(project, serviceName, username string) (*Servi
 	return nil, err
 }
 
+// Put modifies the given Service User in Aiven.
+func (h *ServiceUsersHandler) Put(project, service, username string, password string) (*ServiceUser, error) {
+	msr := ModifyServiceUserRequest{Operation: "reset-credentials", NewPassword: password, Authentication: "caching_sha2_password"}
+	path := buildPath("project", project, "service", service, "user", username)
+	svc, err := h.client.doPutRequest(path, msr)
+	if err != nil {
+		return nil, err
+	}
+	var r ServiceResponse
+	errR := checkAPIResponse(svc, &r)
+	for _, user := range r.Service.Users {
```

Comment:
`r` could be `nil` in the cause of an error and calling of `r.Service.Users` can `panic`. 

I would recommend adding error checking first before iterating over the service users, or just make sure that `r` is not `nil`.